packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -121,13 +126,18 @@ public void onMethodCall(MethodCall call, Result result) {
     if (call.argument("arguments") != null) {
       intent.putExtras(convertArguments((Map) call.argument("arguments")));
     }
-    Log.i("android_intent plugin", "Sending intent " + intent);
-    if (mRegistrar.activity() != null) {
-      mRegistrar.activity().startActivity(intent);
-    } else {
-      mRegistrar.context().startActivity(intent);
+    if (call.argument("package") != null) {
+      intent.setPackage((String) call.argument("package"));
+    }
+    // If we cannot resolve an explicit intent fallback to an implicit one
+    if (intent.getPackage() != null
```

Comment:
Can we merge this conditional with the previous one like so:
```java
if (call.argument("package") != null) {
  intent.setPackage((String) call.argument("package"));
  if (intent.resolveActivity(context.getPackageManager()) == null) {
    intent.setPackage(null);
  }
}
```

Reply:
Yes. Good catch, easier to reason about.